packages/spectral/src/serverTypes/convertComponent.ts

```diff
@@ -36,7 +36,7 @@ export const convertInput = (
     collection === "keyvaluelist" && typeof label === "object" ? label.key : undefined;
 
   return {
-    ...omit(rest, ["onPremControlled"]),
+    ...omit(rest, ["onPremControlled", "permissionAndVisibilityType", "visibleToOrgDeployer"]),
```

Comment:
When you make the updates indicated below this won't be needed anymore since this shouldn't affect component authoring.

Reply:
Had to do it this way because the `convertComponent` takes the CNI definition which then passes it to Prism to run `publishComponent`. `permissionAndVisibilityType` and `visibleToOrgDeployer` are invalid values creating a `Error: Bad Request`.   

```
export const configPages = {
  Connections: configPage({
    elements: {
      brotatoConnection: connectionConfigVar({
        stableKey: "sdf",
        dataType: "connection",
        connection: {
          component: "brotato",
          key: "myConnection",
          values: {
            password: {
              permissionAndVisibilityType: "organization",
              visibleToOrgDeployer: true,
              value: "aaaaa",
            },
            username: {
              permissionAndVisibilityType: "customer",
              visibleToOrgDeployer: false,
              value: "bbbbbb",
            },
          },
        },
      }),
      connection1: connectionConfigVar({
        stableKey: "f675ec15-a95a-42f0-821c-e8546bc78e11",
        oauth2Type: OAuth2Type.AuthorizationCode,
        dataType: "connection",
        permissionAndVisibilityType: "customer",
        inputs: {
          authorizeUrl: {
            label: "Authorize URL",
            placeholder: "Authorize URL",
            type: "string",
            required: false,
            shown: true,
            default: "https://example.com/oauth2/authorize",
            comments: "The OAuth 2.0 Authorization URL for the API",
            permissionAndVisibilityType: "organization",
            visibleToOrgDeployer: false,
          },
          tokenUrl: {
            label: "Token URL",
            placeholder: "Token URL",
    ....
```

**Error:**
```
Error: Bad Request
    Variable '$connections' got invalid value {'placeholder': 'Authorize URL', 'required': False, 'shown': True, 'comments': 'The OAuth 2.0 Authorization URL for the API', 'permissionAndVisibilityType': 'organization', 'visibleToOrgDeployer': False, 'key': 'authorizeUrl', 'type': 'string', 'default': 
    'https://example.com/oauth2/authorize', 'label': 'Authorize URL'} at 'connections[0].inputs[0]'; Field 'permissionAndVisibilityType' is not defined by type 'ConnectionInputFieldDefinition'.
    Variable '$connections' got invalid value {'placeholder': 'Authorize URL', 'required': False, 'shown': True, 'comments': 'The OAuth 2.0 Authorization URL for the API', 'permissionAndVisibilityType': 'organization', 'visibleToOrgDeployer': False, 'key': 'authorizeUrl', 'type': 'string', 'default': 
    'https://example.com/oauth2/authorize', 'label': 'Authorize URL'} at 'connections[0].inputs[0]'; Field 'visibleToOrgDeployer' is not defined by type 'ConnectionInputFieldDefinition'.
    Variable '$connections' got invalid value {'placeholder': 'Token URL', 'required': False, 'shown': True, 'comments': 'The OAuth 2.0 Token URL for the API', 'permissionAndVisibilityType': 'embedded', 'visibleToOrgDeployer': False, 'key': 'tokenUrl', 'type': 'string', 'default': 'asdf', 'label': 'Token 
    URL'} at 'connections[0].inputs[1]'; Field 'permissionAndVisibilityType' is not defined by type 'ConnectionInputFieldDefinition'.
    Variable '$connections' got invalid value {'placeholder': 'Token URL', 'required': False, 'shown': True, 'comments': 'The OAuth 2.0 Token URL for the API', 'permissionAndVisibilityType': 'embedded', 'visibleToOrgDeployer': False, 'key': 'tokenUrl', 'type': 'string', 'default': 'asdf', 'label': 'Token 
    URL'} at 'connections[0].inputs[1]'; Field 'visibleToOrgDeployer' is not defined by type 'ConnectionInputFieldDefinition'.
    Variable '$connections' got invalid value {'placeholder': 'Scopes', 'required': False, 'shown': True, 'comments': 'Space separated OAuth 2.0 permission scopes for the API', 'permissionAndVisibilityType': 'customer', 'visibleToOrgDeployer': False, 'key': 'scopes', 'type': 'string', 'default': 'openid 
    profile email', 'label': 'Scopes'} at 'connections[0].inputs[2]'; Field 'permissionAndVisibilityType' is not defined by type 'ConnectionInputFieldDefinition'.
    Variable '$connections' got invalid value {'placeholder': 'Scopes', 'required': False, 'shown': True, 'comments': 'Space separated OAuth 2.0 permission scopes for the API', 'permissionAndVisibilityType': 'customer', 'visibleToOrgDeployer': False, 'key': 'scopes', 'type': 'string', 'default': 'openid 
    profile email', 'label': 'Scopes'} at 'connections[0].inputs[2]'; Field 'visibleToOrgDeployer' is not defined by type 'ConnectionInputFieldDefinition'.
```